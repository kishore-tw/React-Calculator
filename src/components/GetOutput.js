import React,  { useEffect } from 'react'
import Calculator from './Calculator'

function GetOutput({ inputOne, inputTwo, output, hasError, setInputValues, setOutput}) {
    return(
    <div>
    <Calculator inputOne={inputOne} inputTwo={inputTwo} output= {output} hasError = {hasError} 
    onValueChange={setInputValues} onClickChange={setOutput} />
    </div>)
}

export default GetOutput;