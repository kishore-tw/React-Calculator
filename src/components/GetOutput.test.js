import React from 'react'
import renderer from 'react-test-renderer';
import GetOutput from './GetOutput'

test('should render get output component', () => {
    const getOutput = renderer.create(<GetOutput inputOne={2} inputTwo={3} output= {5} hasError = {true}/>);
    expect(getOutput).toMatchSnapshot();
})