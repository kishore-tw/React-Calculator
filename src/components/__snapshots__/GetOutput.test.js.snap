// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render get output component 1`] = `
<div>
  <form>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root formBox"
      >
        <div
          className="MuiGrid-root"
        >
          <div
            className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
            role="alert"
          >
            <div
              className="MuiAlert-icon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              className="MuiAlert-message"
            >
              Please fill both the fields
            </div>
          </div>
          <div
            className="MuiGrid-root"
          >
            <br />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline input"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              id="input_one"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Input One"
              type="text"
              value={2}
            />
          </div>
        </div>
        <div
          className="MuiGrid-root"
        >
          <br />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline input"
            onClick={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input"
              id="input_two"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Input Two"
              type="text"
              value={3}
            />
          </div>
        </div>
        <div
          className="MuiGrid-root"
        >
          <br />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <span
            className="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              id="plus"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                 + 
              </span>
            </button>
          </span>
          <span
            className="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              id="minus"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                 - 
              </span>
            </button>
          </span>
          <span
            className="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              id="multiply"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                 * 
              </span>
            </button>
          </span>
          <span
            className="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              id="divide"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                 / 
              </span>
            </button>
          </span>
        </div>
        <div
          className="MuiGrid-root"
        >
          <br />
          <br />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            id="clear"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
               Clear 
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root"
        >
          <br />
        </div>
        <div
          className="MuiGrid-root"
        >
          <br />
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root textField"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink={true}
            htmlFor="output"
            id="output-label"
          >
            Output
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              disabled={false}
              id="output"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={true}
              required={false}
              type="number"
              value={5}
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
              >
                <span>
                  Output
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
